packages/front-end/enterprise/components/MetricGroupInlineForm.tsx

```diff
@@ -0,0 +1,91 @@
+import { useForm } from "react-hook-form";
+import { Flex } from "@radix-ui/themes";
+import { MetricGroupInterface } from "back-end/types/metric-groups";
+import { useState } from "react";
+import { useAuth } from "@/services/auth";
+import Field from "@/components/Forms/Field";
+import Button from "@/components/Radix/Button";
+import Callout from "@/components/Radix/Callout";
+import track from "@/services/track";
+
+export default function MetricGroupInlineForm({
+  selectedMetricIds,
+  datasource,
+  mutateDefinitions,
+  onChange,
+  cancel,
+}: {
+  datasource: string;
+  selectedMetricIds: string[];
+  mutateDefinitions: () => Promise<void>;
+  onChange: (metrics: string[]) => void;
+  cancel: () => void;
+}) {
+  const [error, setError] = useState<string | undefined>(undefined);
+  const [loading, setLoading] = useState(false);
+  const { apiCall } = useAuth();
+  const form = useForm<{ name: string; metrics: string[]; datasource: string }>(
+    {
+      defaultValues: {
+        name: "",
+        datasource,
+        metrics: selectedMetricIds,
+      },
+    }
+  );
+
+  const submit = form.handleSubmit(async () => {
+    setError(undefined);
+    setLoading(true);
+    const value = {
+      name: form.watch("name"),
+      metrics: form.watch("metrics"),
+      datasource: form.watch("datasource"),
+    };
+    track("Create Metric Group", {
+      value,
+      source: "Add New Experiment Modal",
+    });
+    try {
+      const res = await apiCall<{
+        status: 200;
+        metricGroup: MetricGroupInterface;
+      }>("/metric-group", {
+        method: "POST",
+        body: JSON.stringify(value),
+      });
+      await mutateDefinitions();
+
+      onChange([res.metricGroup.id]);
+    } catch (e) {
+      setError(
+        `Unable to save Metric Group. ${
+          e.message ? `Reason:  ${e.message}` : ""
+        }`
```

Comment:
I saw in the screenshot in the PR description an error about invalid data source.  Is that a real error?  If so, can we detect that ahead of time and disable the metric groups prompt so they don't even see this form?

Reply:
Re: The "Invalid Data Source", it is a real error, but I don't think its one that'll actually be hit often. I forced it to happen locally just to get an error state.

The error can happen [here](https://github.com/growthbook/growthbook/blob/084bbba85914dfefc8bc56fb75bdd337edf7f209/packages/back-end/src/routers/metric-group/metric-group.controller.ts#L35), where we look up the Data Source by ID. As we're populating the list of data sources on the front end, this shouldn't happen, but it's possible since `getDataSourceById` can return the Data Source or null.